File: CodingSample/DotNet/Enterprise/Networking/TcpClientTest/DemoApp2/Models/ShopModel.cs
using System.Net.Sockets;

namespace DemoApp.Models;

public class ShopModel(string server)
{
    public async Task<ItemInfo> ReadItemInfoAsync(string name)
    {
        string url = $"http://{server}/shop/";
        using var client = new HttpClient{ BaseAddress = new Uri(url) };
        var response = await client.GetAsync(name);
        if(response.IsSuccessStatusCode)
        {
            string message = await response.Content.ReadAsStringAsync();
            return ItemInfo.Parse(message);
        }
        return default;
    }
}

File: CodingSample/Java/Advanced/Networking/TCPClientTest/demo2/src/main/java/app/models/ShopModel.java
package app.models;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class ShopModel {
    
    private String server;

    public ShopModel(String server) {
        this.server = server;
    }

    public ItemInfo readItemInfo(String name) throws Exception {
        URI address = URI.create("http://" + server + "/shop/" + name);
        var client = HttpClient.newHttpClient();
        var request = HttpRequest.newBuilder(address)
            .GET()
            .header("User-Agent", "SalesClient/1.0")
            .header("Accept", "text/plain")
            .build();
        var response = client.send(request, HttpResponse.BodyHandlers.ofString());
        if(response.statusCode() == 200){
            String message = response.body();
            return ItemInfo.parse(message);
        }
        return new ItemInfo(0, 0);
    }
    
}
